src/strategies/balanced.py

```diff
@@ -116,46 +65,161 @@ def step(self, event_list=[]):
+        "{} - {} over maximum load ({} > {})".format(
+            self.current_time, cs.parent, gc.get_current_load(), gc.cur_max_power))
+
+    return charging_stations, avail_bat_power
+
+
+def add_surplus_to_vehicle(self, cs, gc, vehicle, cs_id, charging_stations):
+    """
+    Add left over energy to vehicle
+
+    :param cs: charging station dict
+    :type cs: dict
+    :param gc: grid connector dict
+    :type gc: dict
+    :param vehicle: vehicle dict
+    :type vehicle: dict
+    :param cs_id: name of the charging station
+    :type cs_id: str
+    :param charging_stations: charging stations
+    :type charging_stations: dict
+    :return: current time and commands of the charging stations
+    :rtype: dict
+    """
+
+    if gc.get_current_load() < 0:
+        # surplus power
+        power = clamp_power(-gc.get_current_load(), vehicle, cs)
+        avg_power = vehicle.battery.load(self.interval, power)['avg_power']
+        charging_stations[cs_id] = gc.add_load(cs_id, avg_power)
+        cs.current_power += avg_power
+    elif (vehicle.get_delta_soc() < 0
+          and vehicle.vehicle_type.v2g
+          and cs.current_power < self.EPS
+          and get_cost(1, gc.cost) > self.PRICE_THRESHOLD):
+        # GC draws power, surplus in vehicle and V2G capable: support GC
+        discharge_power = min(
+            gc.get_current_load(),
+            vehicle.battery.loading_curve.max_power * self.V2G_POWER_FACTOR)
+        target_soc = max(vehicle.desired_soc, self.DISCHARGE_LIMIT)
+        avg_power = vehicle.battery.unload(
+            self.interval, discharge_power, target_soc)['avg_power']
+        charging_stations[cs_id] = gc.add_load(cs_id, -avg_power)
+        cs.current_power -= avg_power
+
+    return charging_stations
+
+
+def load_batteries(self):
+    """
+    Load batteries with balanced strategy
+    """
+
+    for b_id, battery in self.world_state.batteries.items():
+        gc = self.world_state.grid_connectors[battery.parent]
+        if get_cost(1, gc.cost) <= self.PRICE_THRESHOLD:
+            # low price: charge with full power
+            power = gc.cur_max_power - gc.get_current_load()
+            power = 0 if power < battery.min_charging_power else power
+            avg_power = battery.load(self.interval, power)['avg_power']
+            gc.add_load(b_id, avg_power)
+        elif gc.get_current_load() < 0:
+            # surplus energy: charge
+            power = -gc.get_current_load()
+            power = 0 if power < battery.min_charging_power else power
+            avg_power = battery.load(self.interval, power)['avg_power']
+            gc.add_load(b_id, avg_power)
+        else:
+            # GC draws power: use stored energy to support GC
+            bat_power = battery.unload(self.interval,
+                                       gc.get_current_load() / battery.efficiency
+                                       )['avg_power']
+            gc.add_load(b_id, -bat_power)
```

Comment:
Use the functions from strategy.py instead. Also goes for greedy and distributed.

Reply:
these are not in dev yet, that's why I didn't use them..